src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -2770,6 +2800,78 @@ QgsProcessingParameterPoint *QgsProcessingParameterPoint::fromScriptCode( const
   return new QgsProcessingParameterPoint( name, description, definition, isOptional );
 }
 
+QgsProcessingParameterGeometry::QgsProcessingParameterGeometry( const QString &name, const QString &description,
+    const QVariant &defaultValue, bool optional )
+  : QgsProcessingParameterDefinition( name, description, defaultValue, optional )
+{
+
+}
+
+QgsProcessingParameterDefinition *QgsProcessingParameterGeometry::clone() const
+{
+  return new QgsProcessingParameterGeometry( *this );
+}
+
+bool QgsProcessingParameterGeometry::checkValueIsAcceptable( const QVariant &input, QgsProcessingContext * ) const
+{
+  if ( !input.isValid() )
+    return mFlags & FlagOptional;
+
+  if ( input.canConvert<QgsProperty>() )
+  {
+    return true;
+  }
+
+  if ( input.canConvert< QgsGeometry >() )
+  {
+    return true;
+  }
```

Comment:
See above -- we should accept QgsRectangle and QgsPoint values too

Reply:
Done.

Accept also QgsReferencedRectangle and QgsReferencedPointXY.

Reply:
thanks!